simpledotorg/simple-server
# Progress Tab: Update facility-wise monthly reports

**Story card:** [sc-9245](https://app.shortcut.com/simpledotorg/story/9245/populate-monthly-reports)

## Because

Some data points were missing in the current `reporting_facility_state_dimensions`
- monthly_follow_ups_htn_and_dm_all
- monthly_follow_ups_htn_and_dm_female
- monthly_follow_ups_htn_and_dm_male
- monthly_follow_ups_htn_and_dm_transgender
- monthly_registrations_htn_and_dm_all
- monthly_registrations_htn_and_dm_female
- monthly_registrations_htn_and_dm_male
- monthly_registrations_htn_and_dm_transgender

## This addresses

- Rename the `reporting_facility_state_dimensions` to `reporting_facility_monthly_follow_ups_and_registrations`
- Add these columns
  - monthly_follow_ups_htn_and_dm_all
  - monthly_follow_ups_htn_and_dm_female
  - monthly_follow_ups_htn_and_dm_male
  - monthly_follow_ups_htn_and_dm_transgender
  - monthly_registrations_htn_and_dm_all
  - monthly_registrations_htn_and_dm_female
  - monthly_registrations_htn_and_dm_male
  - monthly_registrations_htn_and_dm_transgender

## Test instructions

Test data using metabase
 - I have added the query as a [question](https://metabase.simple.org/question/529-facility-wise-monthly-breakdown-by-gender-and-diagnosis) in mat-view for testing with the live data (copied the [question](https://metabase.bd.simple.org/question/152-facility-wise-monthly-breakdown-by-gender-and-diagnosis) in Bangladesh Metabase as well)
 - This could be an overnight refresh 

Test data using Dashboard
 - Turn on feature flag called `new_progress_tab_v1`
 - Should look similar to the screenshot attached
 - You can now cross-check the data in progress tab and Hypertension or Diabetes reports for the same facility
<img width="246" alt="Screenshot 2022-10-20 at 7 43 34 PM" src="https://user-images.githubusercontent.com/32141642/196973064-8a55c426-f194-4670-bd0a-6a5b1e51aa5f.png">
